salsa_dancing_molecules/simulations/nve.py

```diff
@@ -35,16 +37,26 @@ def run_for_materials(formula, api_key, steps, output_path, repeat=0):
         if repeat > 0:
             atoms = atoms.repeat(repeat)
 
-        try:
-            print(f'Simulating for {symbols}')
-            run(atoms, steps, output_name)
-        except AsapError as e:
-            print(f'ASAP3 error for {symbols}: {e}')
-            print('Trying without ASAP3...')
-            run(atoms, steps, output_name, False)
-
-
-def run(atoms, steps, output_path, use_asap=True):
+        while True:
+            try:
+                print(f'Simulating for {symbols}')
+                run(atoms, steps, output_name, potential)
+            except ValueError as e:
+                print(f'Size error for {symbols}: {e}')
+                print('Increasing amount of cells')
+                repeat = repeat + 1
+                atoms = atoms.repeat(repeat)
+            except AsapError as e:
+                print(f'ASAP3 error for {symbols}: {e}')
+                print('Trying without ASAP3...')
+                run(atoms, steps, output_name, 'Lennard-Jones', False)
+                break
+            # else will only be executed if there is no error
+            else:
```

Comment:
Can you add message or print a description of the error here?

Reply:
Both of the errors are covered by the two exceptions which will print information. Is there any error that I am missing?

Reply:
Oh I see, then I'll resolve this conversation.